internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -22,19 +22,18 @@ type UploadOpts struct {
 	UploadFn   UploadFunc // Custom implementation on how to upload the contents under a file. Defaults to S3UploadFn.
 }
 
-// Upload uploads static assets to Cloud Storage.
+// Upload uploads static assets to Cloud Storage and returns uploaded file URLs.
 func Upload(fs afero.Fs, source, destination string, opts *UploadOpts) ([]string, error) {
 	matcher := buildCompositeMatchers(buildReincludeMatchers(opts.Reincludes), buildExcludeMatchers(opts.Excludes))
-	var paths []string
-	pathsPtr := &paths
-	if err := afero.Walk(fs, source, walkFnWithMatcher(pathsPtr, matcher)); err != nil {
-		return nil, fmt.Errorf("walk the file tree rooted at %s: %w", source, err)
+	var urls []string
+	urlsPtr := &urls
+	if err := afero.Walk(fs, source, walkFn(source, destination, fs, opts.UploadFn, urlsPtr, matcher)); err != nil {
```

Comment:
I think this should work 💭 :
```suggestion
	var urls []string
	if err := afero.Walk(fs, source, walkFn(source, destination, fs, opts.UploadFn, &urls, matcher)); err != nil {
```